File: external-libs/android-core/src/main/java/com/serhii/core/security/impl/crypto/Result.java
package com.serhii.core.security.impl.crypto;

public class Result {

    private String message;
    private byte[] iv;
    private CryptoError error;

    public Result(String message, byte[] iv, CryptoError error) {
        this.message = message;
        this.iv = iv;
        this.error = error;
    }

    public String getMessage() {
        return message;
    }

    public byte[] getIv() {
        return iv;
    }

    public CryptoError getError() {
        return error;
    }

    public boolean isResultAvailable() {
        return error == CryptoError.OK;
    }

    public void setIv(byte[] iv) {
        this.iv = iv;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public void setError(CryptoError error) {
        this.error = error;
    }

    @Override
    public String toString() {
        return "";
    }
}

File: external-libs/android-core/src/main/java/com/serhii/core/security/impl/crypto/Result.kt
/*
 * Copyright 2022. Happy coding ! :)
 * Author: Serhii Butryk
 */
package com.serhii.core.security.impl.crypto

import com.serhii.core.security.Crypto

data class Result(
                  // Can contain IV data part
                  val message: String = "",
                  val iv: String = "",
                  val error: CryptoError = CryptoError.UNKNOWN,
                  val hasIV: Boolean = false) {

    // A message without IV data part
    val realMessage = if (hasIV && message.isNotEmpty())
        message.substring(Crypto.IV_SIZE) else message

    val errorOk: Boolean
        get() = error === CryptoError.OK

    override fun toString(): String {
        return ""
    }

    // Implemented for usage in Kotlin collections
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Result

        if (message != other.message) return false
        if (!iv.contentEquals(other.iv)) return false
        if (error != other.error) return false
        if (errorOk != other.errorOk) return false

        return true
    }

    // Implemented for usage in Kotlin collections
    override fun hashCode(): Int {
        var result = message.hashCode()
        result = 31 * result + iv.hashCode()
        result = 31 * result + error.hashCode()
        return result
    }

}